src/formatDate.js

```diff
@@ -7,8 +7,40 @@
  *
  * @returns {string}
  */
+
+const getObjectDate = (dateParts, fromFormat) =>
+  fromFormat.reduce((objectFormat, format, i) => {
+    if (format === 'YY' && dateParts[i].length === 2) {
+      objectFormat['YYYY'] =
+        dateParts[i] < '30' ? `20${dateParts[i]}` : `19${dateParts[i]}`;
+    } else {
+      objectFormat[format] = dateParts[i];
+    }
+
+    return objectFormat;
+  }, {});
```

Comment:
The `getObjectDate` function is well-implemented. However, the check `dateParts[i].length === 2` inside the `if` statement is unnecessary because the input is always correctly formatted, as per the task description. You can safely assume that when 'YY' is encountered in `fromFormat`, `dateParts[i]` will have a length of 2.